src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -602,8 +610,15 @@ module.exports = class LevelPlane {
     for (let i = blocksPositions.length - 1; i >= 0; --i) {
       let destination = [blocksPositions[i][0] + offsetX, blocksPositions[i][1] + offsetY];
       let block = this.getBlockAt(blocksPositions[i]);
-      if (this.inBounds(destination) && this.getBlockAt(destination).isDestroyableUponPush()) {
-        this.levelModel.controller.levelView.playExplosionAnimation(destination, 2, destination, block.blockType, null, null, this.player);
+      if (block.isDestroyableUponPush() || (this.inBounds(destination) && this.getBlockAt(destination).isDestroyableUponPush())) {
+        let newBlock = undefined;
+        if (block.isDestroyableUponPush()) {
```

Comment:
This seems a bit redundant. Why not

```javascript
if (block.isDestroyableUponPush()) {
  // destroy block
} else if (this.inBounds(destination) && this.getBlockAt(destination).isDestroyableUponPush()) {
  // destroy destination
}
```

Reply:
yeah... it's not pretty. For some reason, redstoneTorches were causing stack overflows and redstoneWire weren't. Something to do with being a power source? I'm not sure. 

the first block. check is for the torch, the second destination check is for the wire. I can try and clean it up, but right now it -is- working without crashing the game, which was what was happening in a number of the levels 😱